pre_commit/clientlib.py

```diff
@@ -112,6 +113,32 @@ def validate_manifest_main(argv: Optional[Sequence[str]] = None) -> int:
 META = 'meta'
 
 
+# should inherit from cfgv.Conditional if sha support is dropped
+class CheckBadRev(cfgv.ConditionalOptional):
+    def check(self, dct: Dict[str, Any]) -> None:
+        super().check(dct)
+
+        if self.key in dct.keys():
```

Comment:
`.keys()` should almost never be called -- there's only one valid use case that I've ever seen for it (set operations (`|` / `&`) against keys) and every other use can be replaced by removing `.keys()`, you want `if self.key in dct`